### PR TITLE
Tweak: AI - Add a link to the prompt library when it's relevant [ED-12951]

### PR DESCRIPTION
**Pull request by DennisNerush**
# Description
Add a teaser to use the prompt library from the different prompt types

### CODE REVIEW

File: modules/ai/assets/js/editor/components/prompt-library-link.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Typography, Link } from '@elementor/ui';
+import { __ } from '@wordpress/i18n';
+import PropTypes from 'prop-types';
+
+const PromptLibraryLink = ( props ) => {
+	return (
+		<Typography variant="body2" color="text.secondary">
+			{ __( 'For more inspiration, try experimenting with proven prompts from our ' ) }
```
Comment (matipojo) [inline modules/ai/assets/js/editor/components/prompt-library-link.js:8]: Could you please remove the trailing space from the translated text?
```suggestion
			{ __( 'For more inspiration, try experimenting with proven prompts from our' ) } { ' ' }
```

File: modules/ai/assets/js/editor/components/prompt-library-link.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+			<Link href={ props.libraryLink } className="elementor-clickable">
+				{ __( 'prompt library' ) }
```
Comment (matipojo) [inline modules/ai/assets/js/editor/components/prompt-library-link.js:10]: Is there a need for a space before?
```suggestion
				{ ' ' } { __( 'prompt library' ) }
```

File: modules/ai/assets/js/editor/pages/form-layout/components/prompt-autocomplete.js [hunk 3]
```diff
@@ -29,7 +31,10 @@ TextInput.propTypes = {
 };
 
 const PaperComponent = function( props ) {
+	const { mode } = useConfig();
+	const libraryLink = MODE_VARIATION === mode ? 'https://go.elementor.com/ai-prompt-library-variations/' : 'https://go.elementor.com/ai-prompt-library-containers/';
 	return (
+
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-layout/components/prompt-autocomplete.js:37]:
```suggestion
```

File: modules/ai/assets/js/editor/pages/form-layout/components/prompt-autocomplete.js [hunk 4]
```diff
@@ -47,6 +52,10 @@ const PaperComponent = function( props ) {
 			</Typography>
 			<Divider />
 			{ props.children }
+			<Stack sx={ { m: 2 } }>
+				<PromptLibraryLink libraryLink={ libraryLink } />
+			</Stack>
+
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-layout/components/prompt-autocomplete.js:58]:
```suggestion
```

File: modules/ai/assets/js/editor/pages/form-layout/components/prompt-autocomplete.js [hunk 5]
```diff
@@ -86,7 +95,7 @@ const PromptAutocomplete = ( { onSubmit, ...props } ) => {
 			disableClearable
 			open={ showSuggestions }
 			onClose={ () => setShowSuggestions( false ) }
-			onKeyDown={ ( e ) => {
+			OnKeyDown={ ( e ) => {
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-layout/components/prompt-autocomplete.js:98]:
```suggestion
			onKeyDown={ ( e ) => {
```

File: modules/ai/assets/js/editor/pages/form-layout/components/prompt-form.js [hunk 6]
```diff
@@ -153,7 +153,9 @@ const PromptForm = forwardRef( ( {
 			/>
 
 			<GenerateButton disabled={ isGenerateDisabled } />
+
 		</Stack>
+
```
Comment (matipojo) [inline modules/ai/assets/js/editor/pages/form-layout/components/prompt-form.js:158]:
```suggestion
		</Stack>
```

File: tests/jest/unit/modules/ai/assets/js/editor/components/prompt-library-link.test.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { render } from '@testing-library/react';
+import PromptLibraryLink from '../../../../../../../../../modules/ai/assets/js/editor/components/prompt-library-link';
```
Comment (matipojo) [inline tests/jest/unit/modules/ai/assets/js/editor/components/prompt-library-link.test.js:2]: Try 
```suggestion
import PromptLibraryLink from 'elementor/modules/ai/assets/js/editor/components/prompt-library-link';
```

File: tests/jest/unit/modules/ai/assets/js/editor/components/prompt-library-link.test.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+describe( 'PromptLibraryLink', () => {
+	it( 'Should render a link to the library from props', async () => {
+		const { getByRole } = render(
+			<div data-testid="test_id">
```
Comment (matipojo) [inline tests/jest/unit/modules/ai/assets/js/editor/components/prompt-library-link.test.js:7]: Is the `testid` needed?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/jest/unit/modules/ai/assets/js/editor/components/prompt-library-link.test.js (hunk 8 above)
Comment (DennisNerush) [reply tests/jest/unit/modules/ai/assets/js/editor/components/prompt-library-link.test.js:7]: no, removed the wrapping div all together

**Review by matipojo — CHANGES_REQUESTED** [review]

File: modules/ai/assets/js/editor/components/prompt-library-link.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import { __ } from '@wordpress/i18n';
+import PropTypes from 'prop-types';
+
+const PromptLibraryLink = ( props ) => {
+	return (
+		<Typography variant="body2" color="text.secondary">
+			{ __( 'For more inspiration, try experimenting with proven prompts from our' ) }
+			<Link href={ props.libraryLink } className="elementor-clickable">
+				{ ' ' }{ __( 'prompt library' ) }
+			</Link>
```
[the remark this answers is not in the document]
Comment (DennisNerush) [reply modules/ai/assets/js/editor/components/prompt-library-link.js]: fixe

**Review by matipojo — APPROVED** [review]

**Pull request merged**